doc/make.py

```diff
@@ -1,232 +0,0 @@
-            + options):
-        raise SystemExit("Building HTML failed.")
-
-    # Clean out PDF files from the _images directory
-    for filename in glob.glob('build/%s/_images/*.pdf' % buildername):
-        os.remove(filename)
-
-
-def htmlhelp():
-    """Build Sphinx 'htmlhelp' target. """
-    html(buildername='htmlhelp')
-    # remove scripts from index.html
-    with open('build/htmlhelp/index.html', 'r+') as fh:
-        content = fh.read()
-        fh.seek(0)
-        content = re.sub(r'<script>.*?</script>', '', content,
-                         flags=re.MULTILINE | re.DOTALL)
-        fh.write(content)
-        fh.truncate()
-
-
-def latex():
-    """Build Sphinx 'latex' target. """
-    check_build()
-    # figs()
-    if sys.platform != 'win32':
-        # LaTeX format.
-        if subprocess.call(
-                [sys.executable]
-                + '-msphinx -b latex -d build/doctrees . build/latex'.split()):
-            raise SystemExit("Building LaTeX failed.")
-
-        # Produce pdf.
-        # Call the makefile produced by sphinx...
-        if subprocess.call("make", cwd="build/latex"):
-            raise SystemExit("Rendering LaTeX failed with.")
-    else:
-        print('latex build has not been tested on windows')
-
-
-def texinfo():
-    """Build Sphinx 'texinfo' target. """
-    check_build()
-    # figs()
-    if sys.platform != 'win32':
-        # Texinfo format.
-        if subprocess.call(
-                [sys.executable]
-                + '-msphinx -b texinfo -d build/doctrees . build/texinfo'.split()):
-            raise SystemExit("Building Texinfo failed.")
-
-        # Produce info file.
-        # Call the makefile produced by sphinx...
-        if subprocess.call("make", cwd="build/texinfo"):
-            raise SystemExit("Rendering Texinfo failed with.")
-    else:
-        print('texinfo build has not been tested on windows')
-
-
-def clean():
-    """Remove generated files. """
-    shutil.rmtree("build", ignore_errors=True)
-    shutil.rmtree("tutorials", ignore_errors=True)
-    shutil.rmtree("api/_as_gen", ignore_errors=True)
-    for pattern in ['_static/matplotlibrc',
-                    '_templates/gallery.html',
-                    'users/installing.rst']:
-        for filename in glob.glob(pattern):
-            if os.path.exists(filename):
-                os.remove(filename)
-
-
-def build_all():
-    """Build Sphinx 'html' and 'latex' target. """
-    # figs()
-    html()
-    latex()
-
-
-funcd = {
-    'html':      html,
-    'htmlhelp':  htmlhelp,
-    'latex':     latex,
-    'texinfo':   texinfo,
-    'clean':     clean,
-    'all':       build_all,
-    'doctest':   doctest,
-    'linkcheck': linkcheck,
-    }
-
-
-small_docs = False
-warnings_as_errors = True
-n_proc = 1
-
-# Change directory to the one containing this file
-current_dir = os.getcwd()
-os.chdir(os.path.dirname(os.path.join(current_dir, __file__)))
```

Comment:
I suspect that this line is the source of the import failures.

Reply:
The old make.py compiles fine without those lines, so I don't think that this is the problem.

Reply:
I've opened a PR on sphinx-gallery that "fixes" the problem. I have no clue how the old make.py managed to build the examples considering the sphinx-gallery code I patched.

I would not be surprised that my patch gets rejected, as it opens the door to examples that are not self contained (which is our problem). I think the correct solution would be to make the units examples self-contained and easier to understand.